### PR TITLE
changed the way we store node filenames

### PR DESCRIPTION
**Pull request by fabpot**

### CODE REVIEW

File: lib/Twig/Node.php [hunk 1]
```diff
@@ -229,10 +231,18 @@ public function getIterator()
...
     }
 
-    public function getFilename()
+    public function setFilename($filename)
     {
-        if ($this->hasAttribute('module_filename')) {
-            return $this->getAttribute('module_filename');
+        $this->filename = $filename;
+        foreach ($this->nodes as $node) {
+            if (null !== $node) {
```
Comment (stof) [inline lib/Twig/Node.php]: can it really be null ? This would break the contract

Comment (fabpot) [inline lib/Twig/Node.php]: `parent` is such an example on the main Module node. I agree that this is not great, and probably something I will fix in 2.0.

Comment (fabpot) [inline lib/Twig/Node.php]: Actually, there are more than just `parent` :( I will address that in another PR.

File: lib/Twig/Node.php [hunk 2]
```diff
@@ -229,10 +231,18 @@ public function getIterator()
...
+                $node->setFileName($filename);
```
Comment (xabbuh) [inline lib/Twig/Node.php]: typo: `setFilename()`

**Comment by xabbuh** [comment]
👍 much better solution

File: lib/Twig/Node.php (hunk 1 above)
Comment (fabpot) [inline lib/Twig/Node.php]: see #2123

**Pull request merged**